### PR TITLE
Validate dotenv, simplify dev setup

### LINKED ISSUE #373: Validate dotenv (+ set defaults to simplify dev setup)

### PR DESCRIPTION
**Pull request by Benjamin-Frost**
Closes #373

- [x] I have read the [code-style guidelines](https://github.com/hpi-dhc/PharMe/blob/main/docs/FLUTTER_STYLE.md) and confirmed that this PR upholds said guidelines

### CODE REVIEW

File: .github/workflows/lab-server.yml [hunk 1]
```diff
@@ -16,11 +16,11 @@ env:
...
-  LAB_SERVER_DB_HOST: postgres
-  LAB_SERVER_DB_PORT: 5432
-  LAB_SERVER_DB_USER: postgres
-  LAB_SERVER_DB_PASS: postgres
-  LAB_SERVER_DB_NAME: lab_server_db
+  DB_HOST: postgres
+  DB_PORT: 5432
+  DB_USER: postgres
+  DB_PASS: postgres
+  DB_NAME: lab_server_db
```
Comment (Toalaah) [inline .github/workflows/lab-server.yml:23]:
```suggestion

```

File: .github/workflows/lab-server.yml [hunk 2]
```diff
@@ -52,9 +52,9 @@ jobs:
       postgres:
         image: postgres
         env:
-          POSTGRES_USER: ${{ env.LAB_SERVER_DB_USER }}
-          POSTGRES_PASSWORD: ${{ env.LAB_SERVER_DB_PASS }}
-          POSTGRES_DB: ${{ env.LAB_SERVER_DB_NAME }}
+          POSTGRES_USER: ${{ env.DB_USER }}
+          POSTGRES_PASSWORD: ${{ env.DB_PASS }}
+          POSTGRES_DB: ${{ env.DB_NAME }}
```
Comment (Toalaah) [inline .github/workflows/lab-server.yml:57]:
```suggestion
          POSTGRES_USER: admin
          POSTGRES_PASSWORD: admin
          POSTGRES_DB: lab_server_db
```

Makes it consistent with the annotation server CI file.

File: annotation-server/.env.example [hunk 3]
```diff
@@ -1,24 +1,27 @@
-# Database Config
-ANNOTATION_DB_HOST=localhost
-ANNOTATION_DB_PORT=5432
-ANNOTATION_DB_USER=changeme
-ANNOTATION_DB_PASS=changeme
-ANNOTATION_DB_NAME=annotation_db
+########## Required for Development ##########
+
+# Google Sheets
```
Comment (Toalaah) [inline annotation-server/.env.example:3]: Is `EMPTY_GOOGLE_SHEET_ID` not also required if I want to be able to test my code locally?

File: lab-server/.env.example [hunk 4]
```diff
@@ -1,33 +1,31 @@
...
-# hosts that need to be changed)
+########## Required for Development ##########
 
-# Lab server database (local setup)
-LAB_SERVER_DB_HOST=localhost
-LAB_SERVER_DB_PORT=6543
-LAB_SERVER_DB_USER=postgres
-LAB_SERVER_DB_PASS=postgres
-LAB_SERVER_DB_NAME=lab_server_db
+# Nothing is required for development.
```
Comment (Toalaah) [inline lab-server/.env.example:3]:
```suggestion
```

File: lab-server/.env.example [hunk 5]
```diff
@@ -1,33 +1,31 @@
-# Note: the following env example is configured so that when following the
-# contribution guide for a LOCAL SETUP no values need to be changed (apart from
-# the keycloak lab-server secret which is generated when creating the client in
-# the Keycloak admin panel). This environment will NOT work out of the box when
-# trying to deploy these services. Refer to the docker-compose file when
-# changing your env values to support a deployment (although it's mainly the
-# hosts that need to be changed)
+########## Required for Development ##########
```
Comment (Toalaah) [inline lab-server/.env.example:1]:
```suggestion
```

File: lab-server/.env.example [hunk 6]
```diff
@@ -1,33 +1,31 @@
...
-# Lab server database (local setup)
-LAB_SERVER_DB_HOST=localhost
-LAB_SERVER_DB_PORT=6543
-LAB_SERVER_DB_USER=postgres
-LAB_SERVER_DB_PASS=postgres
-LAB_SERVER_DB_NAME=lab_server_db
+# Nothing is required for development.
 
-# Keycloak credentials (local setup)
+########## Additionally Required for Production ##########
```
Comment (Toalaah) [inline lab-server/.env.example:5]:
```suggestion
########## The following environment variables may need to be adjusted for production environments ##########
```

**Review by Toalaah — CHANGES_REQUESTED** [review]

**Review by Toalaah — APPROVED** [review]

File: lab-server/src/common/validations/env.validation.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    DB_PORT: Joi.number().port().default(6543),
+    DB_USER: Joi.string().default('admin'),
+    DB_PASS: Joi.string().default('admin'),
+    DB_NAME: Joi.string().default('lab_server_db'),
+    // Keycloak
+    KEYCLOAK_USER: Joi.string().default('admin'),
+    KEYCLOAK_PASS: Joi.string().default('admin'),
+    KEYCLOAK_DB_USER: Joi.string().default('admin'),
+    KEYCLOAK_DB_PASS: Joi.string().default('admin'),
+    KEYCLOAK_DB_NAME: Joi.string().default('keycloak_db'),
```
[the remark this answers is not in the document]
Comment (Benjamin-Frost) [reply lab-server/src/common/validations/env.validation.ts:20]: We just removed the local setup which makes this mandatory, no?

[the remark this answers is not in the document]
Comment (Toalaah) [reply lab-server/src/common/validations/env.validation.ts:20]: Pretty sure the vars are only needed for the keycloak service in the docker compose. The lab server should never need the admin credentials to the UI / direct DB access.

[the remark this answers is not in the document]
Comment (Benjamin-Frost) [reply lab-server/src/common/validations/env.validation.ts]: You are completely right, my bad

**Pull request merged**